Source/ARTOSReachability.m

```diff
@@ -0,0 +1,93 @@
+//
+//  ARTReachability.m
+//  Ably
+//
+//  Created by Toni Cárdenas on 2/5/16.
+//  Copyright © 2016 Ably. All rights reserved.
+//
+
+#import <arpa/inet.h>
+#import <ifaddrs.h>
+#import <netdb.h>
+#import <sys/socket.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <SystemConfiguration/SystemConfiguration.h>
+#import <netinet/in.h>
+
+#import "ARTOSReachability.h"
+
+@implementation ARTOSReachability {
+    ARTLog *_logger;
+    NSString *_host;
+    void (^_callback)(BOOL);
+    SCNetworkReachabilityRef _reachabilityRef;
+    NSMutableDictionary *_instances;
+}
+
+- (instancetype)initWithLogger:(ARTLog *)logger {
+    self = [super self];
+    if (self) {
+        _logger = logger;
+        if (ARTOSReachability_instances == nil) {
+            _instances = [[NSMutableDictionary alloc] init];
+            ARTOSReachability_instances = _instances;
+        } else {
+            _instances = ARTOSReachability_instances;
+        }
+    }
+    return self;
+}
+
+__weak NSMutableDictionary *ARTOSReachability_instances;
+
+static void ARTOSReachability_Callback(SCNetworkReachabilityRef target, SCNetworkReachabilityFlags flags, void* info) {
+    [(ARTOSReachability *)ARTOSReachability_instances[[NSValue valueWithPointer:target]] internalCallback:flags & kSCNetworkReachabilityFlagsReachable];
+}
+
+- (void)listenForHost:(NSString *)host callback:(void (^)(BOOL))callback {
+    [self off];
+
+    if (!host) {
```

Comment:
> then comes the _reachability listenForHost:[_transport host] and _transport is nil.

I guess you mean that `[_transport host]` is nil? `_transport` cannot be nil at this point since it's instantiated [right above](https://github.com/ably/ably-ios/pull/444/files/5c2436e3427bb653c71df7f655acf969c6be60fb#diff-d9df428f90fcfee8426c1e46435f952aR250).

Now, why is `[_transport host]` nil? That surely looks like a inconsistent state of `_transport`.